Update windows x64 doc (#3759)
MicrosoftDocs/azure-docs-cli: docs-ref-conceptual/install-azure-cli-windows.md

Download and install the latest release of the Azure CLI. When the installer asks if it can make changes to your computer, click the "Yes" box.

> [!div class="nextstepaction"]
> [Latest release of the Azure CLI (32-bit)](https://aka.ms/installazurecliwindows)

> > [Latest release of the Azure CLI (64-bit)](https://aka.ms/installazurecliwindowsx64)

Reviewer: ```suggestion
> [Latest release of the Azure CLI (64-bit)](https://aka.ms/installazurecliwindowsx64)

If you have previously installed the Azure CLI, running either the 32-bit or 64-bit MSI will overwrite an existing installation.
```

### Specific version

> To download the MSI installer for specific version, change the version segment in URL `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>.msi` (32-bit) or `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>-x64.msi` (64-bit) and download it. Available versions can be found at [Azure CLI release notes](./release-notes-azure-cli.md). (64-bit MSI is available from 2.51.0.)

Author: `-` is missing in `2.51.0x64.msi`.

> To download the MSI installer for specific version, change the version segment in URL `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>.msi` (32-bit) or `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>-x64.msi` (64-bit) and download it. Available versions can be found at [Azure CLI release notes](./release-notes-azure-cli.md). (64-bit MSI is available from 2.51.0.)

Reviewer: > `-` is missing in `2.51.0x64.msi`.

@bebound, is this still an active comment?  I changed line 46 on Thursday so we could get this merged.

> To download the MSI installer for specific version, change the version segment in URL `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>.msi` (32-bit) or `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>-x64.msi` (64-bit) and download it. Available versions can be found at [Azure CLI release notes](./release-notes-azure-cli.md). (64-bit version is available from 2.51.0.)

Reviewer 2: ```suggestion
To download the MSI installer for specific version, change the version segment in URL `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>.msi` (32-bit) or `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>-x64.msi` (64-bit) and download it. Available versions can be found at [Azure CLI release notes](./release-notes-azure-cli.md). (64-bit MSI is available from 2.51.0.)
```

Download and install the latest release of the Azure CLI. When the installer asks if it can make changes to your computer, click the "Yes" box.

> [!div class="nextstepaction"]
> [Latest release of the Azure CLI (32-bit)](https://aka.ms/installazurecliwindows)
> [Latest release of the Azure CLI (64-bit)](https://aka.ms/installazurecliwindowsx64)

If you have previously installed the Azure CLI, running either the 32-bit or 64-bit MSI will overwrite an existing installation.

### Specific version

> To download the MSI installer for a specific version, change the version segment in URL `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>.msi` (32-bit) or `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>-x64.msi` (64-bit) and download it. For example, to install the 32-bit MSI of Azure CLI version [2.51.0](/cli/azure/release-notes-azure-cli#august-01-2023), your URL would be `https://azcliprod.blob.core.windows.net/msi/azure-cli-2.51.0.msi`.  The corresponding 64-bit install would be `https://azcliprod.blob.core.windows.net/msi/azure-cli-2.51.0-x64.msi`.

Reviewer 2: I like the examples as they make the instruction easier to follow.

   $ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; Remove-Item .\AzureCLI.msi
   ```

> This will download and install the latest 32-bit version of the Azure CLI for Windows. If you prefer 64-bit one, please change URL to `https://aka.ms/installazurecliwindowsx64`. If you already have a version installed, the installer will update the existing version.

Reviewer: **version** is again used in this sentence.  If **MSI** doesn't fit all contexts, what about using **32-bit package** or **32-bit installer**?  Anything but **version**, as I agree with @Jiasli (above).

> This will download and install the latest 32-bit version of the Azure CLI for Windows. If you prefer 64-bit one, please change URL to `https://aka.ms/installazurecliwindowsx64`. If you already have a version installed, the installer will update the existing version.

Author: ```suggestion
This will download and install the latest 32-bit installer of the Azure CLI for Windows. If you prefer 64-bit one, please change URL to `https://aka.ms/installazurecliwindowsx64`. If you already have a version installed, the installer will update the existing version.
```

> This will download and install the latest 32-bit installer of the Azure CLI for Windows. If you prefer 64-bit one, please change URL to `https://aka.ms/installazurecliwindowsx64`. If you already have a version installed, the installer will update the existing version.

Reviewer: ```suggestion
This will download and install the latest 32-bit installer of the Azure CLI for Windows. If you prefer a 64-bit install, change URL to `https://aka.ms/installazurecliwindowsx64`.
```

> To install specific version, replace the `-Uri` argument with `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>.msi` (32-bit) or `https://azcliprod.blob.core.windows.net/msi/azure-cli-<version>-x64.msi` (64-bit) with version segment changed. Available versions can be found at [Azure CLI release notes](./release-notes-azure-cli.md). (64-bit MSI is available from 2.51.0.)

Reviewer: ```suggestion
To install a specific version, replace the `-Uri` argument with the URL described in [Specific version](#specific-version).  Here is an example of using the 32-bit installer of the Azure CLI version [2.50.0](/cli/azure/release-notes-azure-cli#july-04-2023) in PowerShell:

   ```PowerShell
   $ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest -Uri https://azcliprod.blob.core.windows.net/msi/azure-cli-2.50.0-x32.msi -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; Remove-Item .\AzureCLI.msi
   ```
```

Author: Could you please remove the `-x32` suffix and I can merge this suggestion.

Reviewer 2: `[Specific version](#specific-version)` is under `# [Microsoft Installer (MSI)](#tab/azure-cli)` tab, but not available under `# [Microsoft Installer (MSI) with Command](#tab/powershell)`, so the link will not be clickable.

   $ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; Remove-Item .\AzureCLI.msi
   ```

This will download and install the latest 32-bit installer of the Azure CLI for Windows. If you prefer a 64-bit install, change URL to `https://aka.ms/installazurecliwindowsx64`. If the Azure CLI is already installed, the installer will overwrite the existing version.

> To install a specific version, replace the `-Uri` argument with the URL described in [Specific version](#specific-version).  Here is an example of using the 32-bit installer of the Azure CLI version [2.51.0](/cli/azure/release-notes-azure-cli#august-01-2023) in PowerShell:

Reviewer 2: `[Specific version](#specific-version)` is under `# [Microsoft Installer (MSI)](#tab/azure-cli)` tab, but not available under `# [Microsoft Installer (MSI) with Command](#tab/powershell)`, so the link will not be clickable.

Author: I change specific version to a include file, please see it works.

Reviewer: In view, the link works just fine.  It is a bit odd taking the customer into another tab, but I'm going to merge this PR.

   ```PowerShell

>    $ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest -Uri https://azcliprod.blob.core.windows.net/msi/azure-cli-2.51.0.msi -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; Remove-Item .\AzureCLI.msi

Reviewer 2: This code block is not closed, causing `# [Windows Package Manager](#tab/winget)` tab to be broken:

![image](https://github.com/MicrosoftDocs/azure-docs-cli/assets/4003950/317d2d15-179c-48cb-b230-db48ddf81ced)

# [Windows Package Manager](#tab/winget)

* `https://aka.ms/`
* `https://azcliprod.blob.core.windows.net/`

> ### Migrate to 64-bit Azure CLI

Reviewer 2: It's better to add a line break after a heading:

```suggestion
### Migrate to 64-bit Azure CLI

```

> Starting from 2.51.0, Azure CLI also provides 64-bit version which is recommended for better performance.

Reviewer 2: ```suggestion
Starting from 2.51.0, Azure CLI also provides 64-bit MSI which is recommended for better performance.
```

If you are using 32-bit version, you can migrate to 64-bit version by following these steps:
1. Check your current CLI version and installed extensions by running `az --version`.

> 2. If the extension folder `%userprofile%\.azure\cliextensions` exists, backup it by renaming. This folder will be created again after migration.

Reviewer 2: "backup" is a noun.
```suggestion
2. If the extension folder `%userprofile%\.azure\cliextensions` exists, make a backup by renaming it. This folder will be created again after migration.
```

> 3. Download and install latest 64-bit version following the [Install or update](#install-or-update) guide. Please note that 32-bit version will be uninstalled automatically.

Reviewer: ```suggestion
3. Download and install latest 64-bit version following the [Install or update](#install-or-update) guide. Please note that 32-bit MSI will be uninstalled automatically.
```

> 4. Install extensions by running `az extension add --name <extension> --version <version>`.

Reviewer: **Question 1:** Our extensions automatically install as 32 or 64 depending on the Core installation?
**Question 2:** When moving up (32-->64) or moving back (64-->32) are all existing extensions automatically reinstalled?
**Question 3:** What happens when a customer is running 64-bit, and for whatever reason, overlays their 64-bit extension folder with their 32-bit extension backup? Is there an expected error message when there is a mismatch?

Reviewer 3: Question 4: Is installation of 64bit on top of 32 bit supported? If not, this needs to be clearly mentioned.

Author: Question 1: Yes
Question 2: No, users have to manually reinstalled the extensions.
Question 3: No, the error will be raised during command execution. We don't know if the dependencies are 32-bit or 64-bit.
Question 4: Yes, 32-bit is uninstalled automatically when installing 64-bit.

Reviewer: > Question 4: Is installation of 64bit on top of 32 bit supported? If not, this needs to be clearly mentioned.

@dcaro, I put a sentence on line 42, and @bebound repeated this information on line 157 and 160.  Let me rework line 27.

> If you have any issue after migration, you can uninstall the 64-bit version and reinstall 32-bit one, then restore the extension folder.

Reviewer: ```suggestion
If you have issues after migration, you can uninstall the 64-bit MSI and reinstall the 32-bit MSI.  If you have made a backup of your 32-bit extension folder, restore your extension folder after the change.
```

* `https://aka.ms/`
* `https://azcliprod.blob.core.windows.net/`

### Migrate to 64-bit Azure CLI

Starting from 2.51.0, Azure CLI also provides 64-bit MSI which is recommended for better performance.

> If you are using 32-bit version, you can migrate to 64-bit version by following these steps:

Reviewer: ```suggestion
Follow these steps to migrate to Azure CLI 64-bit:
```

1. Check your current CLI version and installed extensions by running `az --version`.

> 2. If the extension folder `%userprofile%\.azure\cliextensions` exists, make a backup by renaming it. This folder will be created again after migration.

Reviewer 3: ```suggestion
2. Extensions will need to be reinstalled. It is recommended to perform a backup of the current extension folder `%userprofile%\.azure\cliextensions` by renaming it in case you need to revert to the 32-bit version. This folder will be created again after migration.
```

> 2. Extensions will need to be reinstalled. It is recommended to perform a backup of the current extension folder `%userprofile%\.azure\cliextensions` by renaming it in case you need to revert to the 32-bit version. This folder will be created again after migration.

Reviewer: ```suggestion
1. Extensions will need to be reinstalled. It is recommended to perform a backup of the current extension folder `%userprofile%\.azure\cliextensions` by renaming it in case you choose to revert back to 32-bit. This folder is created automatically when you reinstall an extension.
```

> 3. Download and install latest 64-bit version following the [Install or update](#install-or-update) guide. Please note that 32-bit MSI will be uninstalled automatically.

Reviewer: ```suggestion
1. Download and install latest 64-bit installer as described in [Install or update](#install-or-update). The 32-bit MSI will be automatically uninstalled.
```

> 4. Install extensions by running `az extension add --name <extension> --version <version>`.

Reviewer: ```suggestion
1. Install extensions by running `az extension add --name <extension> --version <version>`.  The Azure CLI also prompts you to install an extension upon first use.
```

Author: Could you please provide more details about the prompt?

> If you have issues after migration, you can uninstall the 64-bit MSI and reinstall the 32-bit MSI.  If you have made a backup of your 32-bit extension folder, restore your extension folder after the change.

Reviewer: ```suggestion
If you have issues after migration, you can uninstall the 64-bit and reinstall the 32-bit MSI.  If you have made a backup of your 32-bit extension folder, restore (rename) your extension folder after the change.
```
